packages/amplify-cli-core/API.md

```diff
@@ -125,6 +125,66 @@ export class AmplifyError extends AmplifyException {
 // @public (undocumented)
 export type AmplifyErrorType = 'AmplifyStudioError' | 'AmplifyStudioLoginError' | 'AmplifyStudioNotEnabledError' | 'ApiCategorySchemaNotFoundError' | 'AuthImportError' | 'BackendConfigValidationError' | 'BucketAlreadyExistsError' | 'BucketNotFoundError' | 'CategoryNotEnabledError' | 'CloudFormationTemplateError' | 'CommandNotSupportedError' | 'ConfigurationError' | 'DeploymentError' | 'DeploymentInProgressError' | 'DirectoryError' | 'DirectoryAlreadyExistsError' | 'DuplicateLogicalIdError' | 'EnvironmentConfigurationError' | 'EnvironmentNameError' | 'EnvironmentNotInitializedError' | 'FeatureFlagsValidationError' | 'FrameworkNotSupportedError' | 'FunctionTooLargeError' | 'InputValidationError' | 'InvalidAmplifyAppIdError' | 'InvalidCustomResourceError' | 'InvalidOverrideError' | 'InvalidStackError' | 'IterativeRollbackError' | 'LambdaLayerDeleteError' | 'MigrationError' | 'MissingAmplifyMetaFileError' | 'MissingOverridesInstallationRequirementsError' | 'ModelgenError' | 'NestedProjectInitError' | 'NoUpdateBackendError' | 'NotImplementedError' | 'OpenSslCertificateError' | 'ParameterNotFoundError' | 'PermissionsError' | 'PluginMethodNotFoundError' | 'PluginNotFoundError' | 'PluginPolicyAddError' | 'ProfileConfigurationError' | 'ProjectAppIdResolveError' | 'ProjectInitError' | 'ProjectNotFoundError' | 'ProjectNotInitializedError' | 'PushResourcesError' | 'RegionNotAvailableError' | 'RemoveNotificationAppError' | 'ResourceAlreadyExistsError' | 'ResourceInUseError' | 'ResourceNotReadyError' | 'StackNotFoundError' | 'StackStateError' | 'UserInputError' | 'MockProcessError' | 'SearchableMockUnsupportedPlatformError' | 'SearchableMockUnavailablePortError' | 'SearchableMockProcessError';
 
+// @public (undocumented)
+export enum AmplifyEvent {
+    // (undocumented)
+    InternalOnlyPostEnvRemove = "InternalOnlyPostEnvRemove",
+    // (undocumented)
+    PostCodegenModels = "PostCodegenModels",
+    // (undocumented)
+    PostEnvAdd = "PostEnvAdd",
+    // (undocumented)
+    PostInit = "PostInit",
+    // (undocumented)
+    PostPull = "PostPull",
+    // (undocumented)
+    PostPush = "PostPush",
+    // (undocumented)
+    PreCodegenModels = "PreCodegenModels",
+    // (undocumented)
+    PreExport = "PreExport",
+    // (undocumented)
+    PreInit = "PreInit",
+    // (undocumented)
+    PrePull = "PrePull",
+    // (undocumented)
+    PrePush = "PrePush"
+}
+
+// @public (undocumented)
+export interface AmplifyEventArgs<T extends AmplifyEvent> {
+    // (undocumented)
+    data?: AmplifyEventData[T];
+    // (undocumented)
+    event: T;
+}
+
+// @public (undocumented)
+export interface AmplifyEventData {
+    // (undocumented)
+    [AmplifyEvent.InternalOnlyPostEnvRemove]: AmplifyInternalOnlyPostEnvRemoveEventData;
```

Comment:
maybe we could just drop `InternalOnly` . if it's public it's anyway wishful thinking.

Reply:
It's referenced by `amplify-cli/`, unfortunately. Looking through the code, I'm going to say let's leave it out of this one. It looks like it might be small ball of yarn.